core/src/test/java/io/nixer/nixerplugin/core/detection/registry/FailedLoginRatioRegistryTest.java

```diff
@@ -22,31 +23,41 @@
 
     private FailedLoginRatioRegistry registry;
 
+    private final Instant baseInstant = Instant.ofEpochSecond(600);
+
+    @BeforeEach
+    void setUp() {
+        registry = new FailedLoginRatioRegistry(Duration.ofMinutes(20), nowSource);
+    }
+
     @Test
     void onDeactivationEvent() {
-        registry = new FailedLoginRatioRegistry(new NowSource());
+        when(nowSource.now()).thenReturn(baseInstant);
+
         registry.onApplicationEvent(new FailedLoginRatioDeactivationEvent(50));
 
         assertThat(registry.isFailedLoginRatioActivated()).isFalse();
     }
 
     @Test
     void onActivationEvent() {
-        registry = new FailedLoginRatioRegistry(new NowSource());
+        when(nowSource.now()).thenReturn(baseInstant);
+
         registry.onApplicationEvent(new FailedLoginRatioActivationEvent(70));
 
         assertThat(registry.isFailedLoginRatioActivated()).isTrue();
     }
 
     @Test
     void activationTimeout() {
-        registry = new FailedLoginRatioRegistry(nowSource);
-        when(nowSource.now()).thenReturn(Instant.now());
+        when(nowSource.now()).thenReturn(baseInstant,
```

Comment:
I know I suggested this configuration by myself, but now I see it might be better to configure `nowSource` mock separately, before each registry invocation, as this would be less coupled to the internal implementation and clearer to understand. 
So the test case would look like below:
```
@Test
    void activationTimeout() {
        when(nowSource.now()).thenReturn(baseInstant);
        registry.onApplicationEvent(new FailedLoginRatioActivationEvent(70));

        when(nowSource.now()).thenReturn(baseInstant.plus(Duration.ofMinutes(5)));
        assertThat(registry.isFailedLoginRatioActivated()).isTrue();

        when(nowSource.now()).thenReturn(baseInstant.plus(Duration.ofMinutes(20)).plusSeconds(1));
        assertThat(registry.isFailedLoginRatioActivated()).isFalse();
}
```

Reply:
No problem, I added comments as well.

Reply:
Thanks!